### PR TITLE
8277369: Strange behavior of JMenuBar with RIGHT_TO_LEFT orientation, arrow keys behaves opposite traversing through keyboard

### PR DESCRIPTION
**Pull request by prsadhuk**
After applying setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT) to the JMenuBar the arrow keys started to behave the opposite of what expected. Right arrow goes left and Left arrow goes right when trying to traverse through keyboard.

Fixed to reverse the direction traversal for RTL orientation.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277369](https://bugs.openjdk.java.net/browse/JDK-8277369): Strange behavior of JMenuBar with RIGHT_TO_LEFT orientation, arrow keys behaves opposite traversing through keyboard

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7396/head:pull/7396` \
`$ git checkout pull/7396`

Update a local copy of the PR: \
`$ git checkout pull/7396` \
`$ git pull https://git.openjdk.java.net/jdk pull/7396/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7396`

View PR using the GUI difftool: \
`$ git pr show -t 7396`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7396.diff">https://git.openjdk.java.net/jdk/pull/7396.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/swing/JMenuBar/MenuBarRTLBug.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    static JMenu forthMenu;
+    static JMenu fifthMenu;
+
+    static Point p;
+    static int width;
+    static int height;
+
+    static volatile boolean passed = false;
+
+    public static void main(String[] args) throws  Exception {
```
Comment (mrserb) [inline test/jdk/javax/swing/JMenuBar/MenuBarRTLBug.java:70]: The submitter said that he use the native L&F should not we check it in the test?

Comment (prsadhuk) [reply test/jdk/javax/swing/JMenuBar/MenuBarRTLBug.java]: Might as well. Added L&F iteration in the test.

**Review by prrace — APPROVED** [review]
Seems to me right should move right and left move left, so OK by me but
I'd like to make sure this is seen by @naotoj too.

**Comment by naotoj** [comment]
Caret should move visually, so VK_RIGHT should move it to the right. Fine by me.

**Review by naotoj — APPROVED** [review]

**Review by mrserb — APPROVED** [review]

**Comment by prrace** [comment]
BTW I was just looking around and if you do a google search on "rtl arrow key backward" it looks to me as if multiple unrelated open source projects happen to have had the same issue (not SFAICT Java based) and came to the same conclusion.

**Pull request closed without merging**